Modules/rental_options.rb

```diff
@@ -0,0 +1,43 @@
+require_relative '../Components/rental'
+
+class RentalOptions
+  attr_accessor :rentals_list
+
+  def initialize(book_options, people_options)
+    @rentals_list = []
+    @book_options = book_options
+    @people_options = people_options
+  end
+
+  def create_rental
+    puts 'Select a book from the following list by number'
+    @book_options.books_list.each_with_index do |book, index|
+      puts "#{index}) Title: '#{book.title}', Author: #{book.author}"
+    end
+    book_num = gets.chomp.to_i
+    puts
+    puts 'Select a person from the following list by number (not id)'
+    @people_options.people_list.each_with_index do |person, index|
+      puts "#{index}) [#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+    person_num = gets.chomp.to_i
+
+    print 'Date: '
+    date = gets.chomp
+    @rentals_list.push(Rental.new(date, @book_options.books_list[book_num], @people_options.people_list[person_num]))
+    puts 'Rental created successfully'
+  end
+
+  def list_rentals
+    puts 'Select a person from the following list by id '
+    @people_options.people_list.each_with_index do |person, index|
+      puts "#{index}) [#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+    id = gets.chomp.to_i
+
+    puts 'Rentals:'
+    @rentals_list.each do |rental|
+      puts "Date: #{rental.date}, Book \"#{rental.book.title}\" by #{rental.book.author}" if rental.person.id == id
```

Comment:
- [ ] it would be better to separate the code that requests and validates inputs from the users like:
```ruby 
book_num = gets.chomp.to_i # code that requests for inputs 
 puts 'Select a person from the following list by number (not id)'
```
and put them in methods and then use the methods inside `create_rental`, `create_book`, `create_person` `list_book`...etc that way you will achieve the single responsibility responsible please try to do that for all the methods